subsys/usb/class/audio/audio.c

```diff
@@ -664,6 +818,97 @@ static int handle_interface_req(struct usb_setup_packet *pSetup,
 	return 0;
 }
 
+/**
+ * @brief Handler for endpoint sampling frequency control requests.
+ *
+ * This function handles endpoint sampling frequency control request.
+ *
+ * @param audio_dev_data USB audio device data.
+ * @param pSetup	 Information about the executed request.
+ * @param len		 Size of the buffer.
+ * @param data		 Buffer containing the request result.
+ *
+ * @return 0 if succesfulf, negative errno otherwise.
+ */
+static int handle_sampling_freq_req(struct usb_audio_dev_data *audio_dev_data,
+				    struct usb_setup_packet *pSetup,
+				    int32_t *len, uint8_t **data, uint8_t ep,
+				    uint8_t device)
+{
+	uint8_t *frequency = audio_dev_data->frequency[device];
+
+	switch (pSetup->bRequest) {
+	case USB_AUDIO_SET_CUR:
+		memcpy(frequency, *data, 3);
+		break;
+	case USB_AUDIO_GET_CUR:
+		memcpy(*data, frequency, 3);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Inform the app */
+	if (audio_dev_data->ops &&
+	    audio_dev_data->ops->sampling_freq_update_cb) {
+		if (pSetup->bRequest == USB_AUDIO_SET_CUR) {
+			enum usb_audio_direction dir;
+			uint32_t freq;
+			freq = frequency[0] << 0;
+			freq += frequency[1] << 8;
+			freq += frequency[2] << 16;
+			dir = (ep & USB_EP_DIR_MASK) ?
+				USB_AUDIO_IN : USB_AUDIO_OUT;
+			audio_dev_data->ops->sampling_freq_update_cb(
+				audio_dev_data->common.dev, dir, freq);
+		}
+	}
+
+
+	return 0;
+}
+
+/**
+ * @brief Handler called for class specific endpoint request.
+ *
+ * This function handles all class specific endpoint requests to a usb audio
+ * device. If request is properly server then 0 is returned. Returning negative
+ * value will lead to set stall on IN EP0.
+ *
+ * @param pSetup    Information about the executed request.
+ * @param len       Size of the buffer.
+ * @param data      Buffer containing the request result.
+ *
+ * @return  0 on success, negative errno code on fail.
+ */
+static int handle_endpoint_req(struct usb_setup_packet *pSetup,
+			       int32_t *len,
+			       uint8_t **data)
+{
+	struct usb_audio_dev_data *audio_dev_data;
+	uint8_t device;
+
+	uint8_t ep = (pSetup->wIndex) & 0xFF;
+	uint8_t cs = ((pSetup->wValue) >> 8) & 0xFF;
+
+	audio_dev_data = get_audio_dev_data_by_ep(ep, &device);
+	if (audio_dev_data == NULL) {
+		LOG_INF("audio_dev_data is NULL");
```

Comment:
Looks like a error, consider to change it ot LOG_ERR or LOG_WRN

Reply:
Ended up removing the error message to be in line with how the custom handler handles it.